### ISSUE TITLE
Could not parse a valid private tree using the given key

### ISSUE THREAD

**Issue by icidasset** [fission-suite/webnative#159]
<img width="445" alt="Screenshot 2021-01-06 at 22 31 00" src="https://user-images.githubusercontent.com/296665/103821662-f9a69280-506e-11eb-9d95-3e8e125e8d49.png">

Happened on Diffuse while trying to write to a file.

**Comment by icidasset** [comment]
Notes:
- Clearing browser data does not resolve the issue.
- Seems to happen to for files too:
<img width="771" alt="Screenshot 2021-01-08 at 16 48 11" src="https://user-images.githubusercontent.com/296665/104034573-58841d00-51d1-11eb-94ee-2e74aa1094ca.png">

**Comment by icidasset** [comment]
I suspect this might have something to do with concurrency, but not sure..
I'm testing with multiples files that are ~3MB

**Comment by dholms** [comment]
:thinking: are you able to consistently replicate this, or does it happen sporadically?
I'm guessing it is something to do with concurrency because I don't remember this happening before that

On this note, I've been meaning to fix this error handling code for a while to give better messages and be a bit more tolerant (we probably shouldn't throw a top level error for one corrupted file)

**Comment by icidasset** [comment]
I can replicate it sometimes on my one filesystem, but not on my other one. And when I did get it to trigger, it was when doing concurrent operations. So it's probably a concurrency issue yeah.

> On this note, I've been meaning to fix this error handling code for a while to give better messages and be a bit more tolerant (we probably shouldn't throw a top level error for one corrupted file)

That'd be great ☺️

**Comment by dholms** [comment]
> I can replicate it sometimes on my one filesystem, but not on my other one.
strange. were they both _created_ with the same version of the FS? or is one older than the other?

**Comment by icidasset** [comment]
Yeah, same filesystem version.

**Comment by dholms** [comment]
ok cool. thanks for flagging this. I wanna improve error handling regardless. And I'll take another look at the concurrency code to see if I can't figure out where this is coming from. Big thanks for wrapping that up & getting it merged too! :raised_hands:

**Comment by icidasset** [comment]
Did some more testing on this, got it to work, but was a very strange process 😅 

- I got this error because I was trying to write to `private/Audio/Music/Playlists/playlist.json`
- I removed `private/Audio/Music/`
- `fs.mkdir("private/Audio/Music/Playlists/")` silently failed (ie. it didn't create the directory)
- Still got the same error
- I removed `private/Audio/`
- `fs.mkdir("private/Audio/Music/Playlists/")` worked fine
- I no longer have the error in question

So I'm guessing when we created the initial directory structure (sample content I think it's called in the code), something went wrong regarding aes keys. This does not always happen, because I tested this on staging and didn't get this error.
